effekt/shared/src/main/scala/effekt/source/Tree.scala

```diff
@@ -112,6 +112,33 @@ case class Comment() extends Tree {
 
 type Doc = Option[String]
 
+/**
+ * The meta-data of a definition / declaration, consisting of everything up
+ * until, but not including the introducing keyword (such as `def`, or `interface`).
+ * For example:
+ *
+ *     /// documentation
+ *     private
+ *     extern
+ *     pure
+ *     def
+ */
+case class Info(
+  doc: Doc,
+  // we use Maybe[Unit] instead of Boolean to have position info for validation errors
+  isPrivate: Maybe[Unit],
+  isExtern: Maybe[Unit],
+  // TODO this should be moved from the Info to extern functions, once we changed the syntax
+  externCapture: Option[CaptureSet]
+) {
```

Comment:
I am also very open to the design of this info object. In order to report some errors on the flags, they are positioned (using `Maybe[Unit]` is a hack). However, an empty `Info` object still contains a lot of position info and thus complicates the tests etc.